google/cloud/storage/async/client.h

```diff
@@ -252,6 +252,66 @@ class AsyncClient {
          std::move(options)});
   }
 
+  /**
+   * Uploads a new object without buffering.
+   *
+   * Use this API to upload objects without any client-side buffering. The API
+   * permits resuming an upload after a transient error. If the application
+   * can checkpoint a simple string the API can even be used to resume an upload
+   * even after the application restarts.
+   *
+   * In both cases this API assumes the application can start sending data from
+   * an arbitrary point in the data source. That makes this API best suited to
+   * upload persistent data sources, such as files, of arbitrary size.
+   *
+   * To upload relatively small objects consider using `InsertObject` instead.
+   * To upload streaming data sources, where rewinding to an arbitrary point
+   * may be impossible consider using `StreamingUpload()`.
+   *
+   * This function always uses [resumable uploads][resumable-link]. The
+   * application can provide a `#RestoreResumableUploadSession()` option to
+   * resume a previously created upload. The returned object has accessors to
+   * query the session id and the next byte expected by GCS.
```

Comment:
nit: here and once below, do we capitalize "ID" ?

Reply:
🤷

Reply:
and done.